text/0001-agile-coretime.md

```diff
@@ -0,0 +1,627 @@
+
+The Coretime-Chain periodically instructs the Relay-chain to assign its cores to alternative tasks as and when Core allocations change due to new Regions coming into effect.
+
+#### Renewal and Migration
+
+There is a renewal system which allows a Bulk Coretime assignment of a single core to be renewed unchanged with a known price increase from month to month. Renewals are processed in a period prior to regular purchases, effectively giving them precedence over a fixed number of cores available.
+
+Renewals are only enabled when a core's assignment does not include an Instantaneous Coretime allocation and has not been split into shorter segments.
+
+Thus, renewals are designed to ensure only that committed parachains get some guarantees about price for predicting future costs. This price-capped renewal system only allows cores to be reused for their same tasks from month to month. In any other context, Bulk Coretime would need to be purchased regularly.
+
+As a migration mechanism, pre-existing leases (from the legacy lease/slots/crowdloan framework) are initialized into the Coretime-chain and cores assigned to them prior to Bulk Coretime sales. In the sale where the lease expires, the system offers a renewal, as above, to allow a priority sale of Bulk Coretime and ensure that the Parachain suffers no downtime when transitioning from the legacy framework.
+
+#### Instantaneous Coretime
+
+Processing of Instantaneous Coretime happens in part on the Polkadot Relay-chain. Credit is purchased on the Coretime-chain for regular DOT tokens, and this results in a DOT-denominated Instantaneous Coretime Credit account on the Relay-chain being credited for the same amount.
+
+Though the Instantaneous Coretime Credit account records a balance for an account identifier (very likely controlled by a collator), it is *non-transferable* and *non-refundable*. It can only be consumed in order to purchase some Instantaneous Coretime with immediate availability.
+
+The Relay-chain reports this usage back to the Coretime-chain in order to allow it to reward the providers of the underlying Coretime, either the Polkadot System or owners of Bulk Coretime who contributed to the Instantaneous Coretime Pool.
+
+Specifically the Relay-chain is expected to be responsible for:
+
+- holding non-transferable, non-refundable DOT-denominated Instantaneous Coretime Credit balance information for collators.
+- setting and adjusting the price of Instantaneous Coretime based on usage.
+- allowing collators to consume their Instantaneous Coretime Credit at the current pricing in exchange for the ability to schedule one PoV for near-immediate usage.
```

Comment:
Probably something like "purchase coretime and use within some short period". The timeframe doesn't matter that much. The two constraints in practice (beyond the scope of the RFC) are:
  1. The economic constraint. Scheduled Coretime needs to expire within a short time period where coretime price volatility is pretty low, or it can be "saved up" when cheap and spent when expensive. This is what happened with GasToken in Ethereum and it caused large problems. This constraint leads to necessary expiry times of hours or days.
  2. The resource constraint. Scheduled coretime shouldn't clog up the scheduler, so the turnaround from buying instantaneous coretime to using it needs to be spent in a timeframe that doesn't overload the scheduler. This constraint probably indicates an expiry time of 30s-1m, depending on what the relay-chain scheduler can handle.